database/seeds/UserBeatmapSeeder.php

```diff
@@ -0,0 +1,350 @@
+                //   echo $e->getMessage();
+                 $exceptions[] = $e->getMessage();
+                }
+            }
+              // END BEATMAPS
+
+              // BEATMAPSETS
+                  // echo '<h3>Beatmap Sets</h3>';
+              $beatmapsets = json_decode(file_get_contents($datapath.'beatmapsets.json'));
+            foreach ($beatmapsets as $beatmapset) {
+                $set = \App\Models\BeatmapSet::where('beatmapset_id', $beatmapset->beatmapset_id)->first();
+                if ($set) {
+                    $set->delete();
+                    $overbeatmapsets[] = $beatmapset->beatmapset_id;
+                  // echo 'Overriding beatmapset: ';
+                }
+                try {
+                    $set = new \App\Models\BeatmapSet;
+                    $set->beatmapset_id = $beatmapset->beatmapset_id;
+                    $set->creator = $beatmapset->creator;
+                    $set->artist = $beatmapset->artist;
+                    $set->title = $beatmapset->title;
+                    $set->displaytitle = $beatmapset->title;
+                    $set->source = $beatmapset->source;
+                    $set->tags = $beatmapset->tags;
+                    $set->bpm = $beatmapset->bpm;
+                    $set->approved = $beatmapset->approved;
+                    $set->approved_date = $beatmapset->approved_date;
+                    $set->genre_id = $beatmapset->genre_id;
+                    $set->language_id = $beatmapset->language_id;
+                    $set->versions_available = $beatmapset->versions_available;
+                    $set->difficulty_names = $beatmapset->difficulty_names;
+                    $set->play_count = $beatmapset->play_count;
+                    $set->favourite_count = $beatmapset->favourite_count;
+
+                    $set->save();
+                  // echo 'Beatmapset '.$beatmapset->title.  ' Saved<br>';
+                  $savedbeatmapsets[] = $beatmapset->beatmapset_id;
+                } catch (\Illuminate\Database\QueryException $e) {
+                    echo ' Error: Unable to save Beatmap Set.';
+                  // echo $e->getMessage();
+                  $exceptions[] = $e->getMessage();
+                }
+            }
+               // END BEATMAPSETS
+
+               // SCORES
+                  //  echo '<h3>Scores</h3>';
+               $scores = json_decode(file_get_contents($datapath.'scores.json'));
+            foreach ($scores as $score) {
+                if ($existing_score = \App\Models\Score\Osu::where('user_id', $score->user_id)->where('beatmap_id', $score->beatmap_id)->first()) {
+                    $existing_score->delete(); // overwrite existing score
+                 $overscores[] = $score->user_id.'-'.$score->beatmap_id;
+                }
+
+                //  echo 'Over-riding: ';
+                     try {
+                         $sc = new \App\Models\Score\Osu;
```

Comment:
this should will achieve same thing without needing to manually assign everything

```
App\Models\Score\Osu::unguard();
new App\Models\Score\Osu($score);
App\Models\Score\Osu::reguard();
```